internal/service/networkpeering/resource_network_peering_test.go

```diff
@@ -176,42 +137,65 @@ func TestAccNetworkRSNetworkPeering_AWSDifferentRegionName(t *testing.T) {
-					resource.TestCheckResourceAttr(resourcePeerName, "vpc_id", vpcID),
-					resource.TestCheckResourceAttr(resourcePeerName, "aws_account_id", awsAccountID),
+				Config: configAWS(orgID, projectName, providerName, vpcID, awsAccountID, vpcCIDRBlock, containerRegion, peerRegion),
+				Check:  resource.ComposeTestCheckFunc(checks...),
+			},
+		},
+	})
+}
 
-					// Container test
-					resource.TestCheckResourceAttrSet(resourceContainerName, "project_id"),
-					resource.TestCheckResourceAttrSet(resourceContainerName, "atlas_cidr_block"),
-					resource.TestCheckResourceAttrSet(resourceContainerName, "provider_name"),
-					resource.TestCheckResourceAttrSet(resourceContainerName, "region_name"),
+func basicAWSTestCase(tb testing.TB) *resource.TestCase {
+	tb.Helper()
+	var (
+		orgID           = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		vpcID           = os.Getenv("AWS_VPC_ID")
+		vpcCIDRBlock    = os.Getenv("AWS_VPC_CIDR_BLOCK")
+		awsAccountID    = os.Getenv("AWS_ACCOUNT_ID")
+		containerRegion = os.Getenv("AWS_REGION")
+		peerRegion      = strings.ToLower(strings.ReplaceAll(containerRegion, "_", "-"))
+		providerName    = "AWS"
+		projectName     = acc.RandomProjectName()
+	)
+	checks := commonChecksAWS(vpcID, providerName, awsAccountID, vpcCIDRBlock, peerRegion)
 
-					resource.TestCheckResourceAttr(resourceContainerName, "project_id", projectID),
-					resource.TestCheckResourceAttr(resourceContainerName, "atlas_cidr_block", "192.168.200.0/21"),
-					resource.TestCheckResourceAttr(resourceContainerName, "provider_name", providerName),
-					resource.TestCheckResourceAttr(resourceContainerName, "region_name", containerRegion),
-				),
+	return &resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(tb); acc.PreCheckPeeringEnvAWS(tb) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             acc.CheckDestroyNetworkPeering,
+		Steps: []resource.TestStep{
+			{
+				Config: configAWS(orgID, projectName, providerName, vpcID, awsAccountID, vpcCIDRBlock, containerRegion, peerRegion),
+				Check:  resource.ComposeTestCheckFunc(checks...),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportStateIdFunc:       importStateIDFunc(resourceName),
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"accepter_region_name", "container_id"},
 			},
 		},
-	})
+	}
+}
+
+func commonChecksAWS(vpcID, providerName, awsAccountID, vpcCIDRBlock, regionPeer string) []resource.TestCheckFunc {
+	attributes := map[string]string{
+		"vpc_id":                 vpcID,
+		"provider_name":          providerName,
+		"aws_account_id":         awsAccountID,
+		"route_table_cidr_block": vpcCIDRBlock,
+	}
+	checks := []resource.TestCheckFunc{checkExists(resourceName, &matlas.Peer{})}
+	checks = acc.AddAttrChecks(resourceName, checks, attributes)
+	checks = append(checks, resource.TestCheckResourceAttr(resourceName, "accepter_region_name", regionPeer))
```

Comment:
Getting some strange behavior here:
- when passing: `us-east-1`
    - `data.mongodbatlas_network_peering.test: Attribute 'accepter_region_name' not found`
- when passing: `US_EAST_1`
  - `Check failed: Check 2/20 error: mongodbatlas_network_peering.test: Attribute 'accepter_region_name' expected "US_EAST_1", got "us-east-1"`
  
  I can see a comment `//Workaround until fix` in https://github.com/mongodb/terraform-provider-mongodbatlas/blob/1015e6206d7addfb2561ed7e4c9bcf6b14fddfb4/internal/service/networkpeering/data_source_network_peering.go#L127
  
  @AgustinBettati what do you think?